Add reusable workflows

Anupamatd opened the pull request.

Anupamatd committed:
Add reusable workflows

danmarsden commented:
looks like a nice start! - good to see this is possible!

Personally I'd prefer to see this simplified a bit more...

In our plugin (eg auth_outage) workflow I'd like to see a config var that defines a list of versions this plugin supports eg something that just holds some branch names:
MOODLE_39_STABLE, MOODLE_310_STABLE, master

and then a one line include of a shared workflow file - the one line shared workflow itself would then figure out which versions of php, which versions of the database and any other dependences we want to normally run..

There's quite a lot of duplication in the custom workflows you've written (one for each branch) - I think we could probably make that easier to maintain by re-using a lot more in a single file?

Anupamatd commented:
Thank you @danmarsden for the comments !!!
Sure, I'll fix this.
Thanks,
Anu

Anupamatd commented:
Hi @danmarsden ,

Implemented below items based on the comment.

* added a config var that a range of branch numbers of supported moodle versions
* called reusable workflow for each branch in a single line
* removed duplication in reusable workflow - https://github.com/catalyst/catalyst-moodle-workflows

Could you please review? 

Thanks,
Anu

brendanheywood commented on version.php:
```diff
@@ -28,7 +28,8 @@
 $plugin->component = "auth_outage";
-$plugin->version = 2021032500;                  // The current plugin version (Date: YYYYMMDDXX).
-$plugin->release = 2021032500;                  // Human-readable release information.
+$plugin->version = 2022011300;                  // The current plugin version (Date: YYYYMMDDXX).
```
make sure these are micro bumps for old stables

2021032501 instead of 
2022011300

Anupamatd replied:
Thank you @brendanheywood - This has been fixed.

Thanks,
Anu
